src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/BaselineHelper.cs

```diff
@@ -101,15 +102,23 @@ public static string RemoveNetTfmPaths(string source)
 
         public static string RemoveVersions(string source)
         {
+            string result = RemoveNetTfmPaths(source);
+
+            // Remove build versions
+            // The semantic version regex incorectly parses through build versions such as "8.1.00-beta.final"
+            // so the build version regex is used to capture matches that the semantic version does not.
+            Regex BuildVersionPattern = new(@"\d+(\.\d+)+([@-](\w+)*\d*([\.+-](alpha|preview|rc|rtm|beta|final|servicing|bundled))*(\.\d+)*)*");
+            result = BuildVersionPattern.Replace(result, VersionPlaceholder);
+
             // Remove semantic versions
             // Regex source: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
             Regex semanticVersionRegex = new(
                 $"(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)"
                 + $"(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))"
                 + $"?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?");
-            string result = semanticVersionRegex.Replace(source, VersionPlaceholder);
+            result = semanticVersionRegex.Replace(result, VersionPlaceholder);
 
-            return RemoveNetTfmPaths(result);
+            return result;
```

Comment:
Consider returning on the last assignment of result instead for simplicity.

Reply:
Updated in this [commit](https://github.com/dotnet/installer/pull/17233/commits/00b01acd87a390755fd426f7dda2d0ae9e1e9efb).